src/App.tsx

```diff
@@ -1,26 +1,274 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  deleteTodo,
+  getTodos,
+  createTodo,
+  USER_ID,
+  updateTodo,
+} from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { Notification } from './components/Notification';
+import { Todo } from './types/Todo';
+import { MESSAGE, ACTION } from './const';
+import { FilterState } from './types/FilterState';
 
-const USER_ID = 0;
+const getFilteredTodo = (todos: Todo[], query: FilterState): Todo[] => {
+  if (query === 'All') {
+    return todos;
+  }
+
+  return todos.filter(
+    todo => todo.completed === (query === FilterState.Completed),
+  );
+};
 
 export const App: React.FC = () => {
+  const [loadingTodoId, setLoadingTodoId] = useState<Todo['id'] | null>(null);
+  const [loadingCompleted, setLoadingCompleted] = useState(false);
+  const [loadingToggles, setLoadingToggle] = useState(false);
```

Comment:
```suggestion
  const [loadingTodoIds, setLoadingTodoIds] = useState<Todo['id'][] | null>(null);
```
we can remove those states and just save the ids of todos on which we need to show the loader